tools/rp_storage_tool/src/fundamental.rs

```diff
@@ -95,6 +95,167 @@ impl<'de> Deserialize<'de> for NTR {
     }
 }
 
+#[derive(Clone, Eq, PartialEq, Hash, Debug, Ord, PartialOrd)]
+pub struct LabeledNTR {
+    pub ntr: NTR,
+    pub label: Option<String>,
+}
+
+impl LabeledNTR {
+    pub fn from_str(s: &str) -> Self {
+        lazy_static! {
+            static ref NTP_MASK_EXPR: Regex =
+                Regex::new("([^]]+/)?([^]]+)/([^]]+)_([^_]+)").unwrap();
+        }
+
+        if let Some(grps) = NTP_MASK_EXPR.captures(&s) {
+            let mut label: Option<String> = None;
+            let label_opt = grps.get(1);
+            if label_opt.is_none() {
+                let mut label_str = label_opt.unwrap().as_str().to_string();
```

Comment:
why do we unwrap a `None`? Should this branch be taken if label opt is `Some`?

Reply:
If the intent is to get the label from the label opt we can do something like

```
let label = grps.get(1).map(|s| s.as_str().chars().take(s.len() - 1).collect())
```

Reply:
Huh. Nice catch! Will fix